File: gitdot-web/packages/api/src/endpoint/repository/update-repository-commit-filter.ts
import { z } from "zod";
import { RepositoryCommitFilterResource } from "../../resource";
import type { Endpoint } from "../endpoint";

export const UpdateRepositoryCommitFilterRequest = z.object({
  name: z.string(),
  authors: z.array(z.string()).optional(),
  tags: z.array(z.string()).optional(),
  paths: z.array(z.string()).optional(),
});
export type UpdateRepositoryCommitFilterRequest = z.infer<
  typeof UpdateRepositoryCommitFilterRequest
>;

export const UpdateRepositoryCommitFilterResponse =
  RepositoryCommitFilterResource;
export type UpdateRepositoryCommitFilterResponse = z.infer<
  typeof UpdateRepositoryCommitFilterResponse
>;

export const UpdateRepositoryCommitFilter = {
  path: "/repository/{owner}/{repo}/commit-filters/{filter_id}",
  method: "PATCH",
  request: UpdateRepositoryCommitFilterRequest,
  response: UpdateRepositoryCommitFilterResponse,
} as const satisfies Endpoint;
export type UpdateRepositoryCommitFilter = typeof UpdateRepositoryCommitFilter;

File: gitdot-api/src/endpoint/repository/update_repository_commit_filter.rs
use serde::{Deserialize, Serialize};

use crate::{endpoint::Endpoint, resource::RepositoryCommitFilterResource};

pub struct UpdateRepositoryCommitFilter;

impl Endpoint for UpdateRepositoryCommitFilter {
    const PATH: &'static str = "/repository/{owner}/{repo}/commit-filters/{filter_id}";
    const METHOD: http::Method = http::Method::PATCH;

    type Request = UpdateRepositoryCommitFilterRequest;
    type Response = UpdateRepositoryCommitFilterResponse;
}

#[derive(ApiRequest, Debug, Serialize, Deserialize)]
pub struct UpdateRepositoryCommitFilterRequest {
    pub name: String,

    #[serde(default, skip_serializing_if = "Option::is_none")]
    pub authors: Option<Vec<String>>,

    #[serde(default, skip_serializing_if = "Option::is_none")]
    pub tags: Option<Vec<String>>,

    #[serde(default, skip_serializing_if = "Option::is_none")]
    pub paths: Option<Vec<String>>,
}

pub type UpdateRepositoryCommitFilterResponse = RepositoryCommitFilterResource;
